### PR TITLE
Add boundary checks and unit tests to abi package

### PR DESCRIPTION
**Pull request by yawangwang**
This PR adds robust boundary checks and size validations to the `abi` package to prevent potential out-of-bounds panics and integer truncation/overflow vulnerabilities when parsing untrusted TDX attestation quotes.

Previously, several internal helper functions in `abi/abi.go` sliced input byte arrays using hardcoded offsets without verifying if the input slice was large enough. This could lead to runtime panics if an attacker provided a malformed or truncated quote.

### 1. Boundary Checks (`abi/abi.go`)
Added explicit slice length validation at the entry point of all internal parsing helpers before any slicing or reading operation occurs:
* **`headerToProto`**: Verifies input is equal to `headerSize` (48 bytes).
* **`tdQuoteBodyToProto`**: Verifies input is equal to `tdQuoteBodySize` (584 bytes).
* **`tdQuoteBodyDescriptorToProto`**: 
  - Verifies input is at least large enough to read the descriptor type and size (6 bytes).
  - Validates `TdQuoteBodyType` early to fail safely on unsupported types before parsing.
  - Verifies the input slice is at least as large as the descriptor's specified `TdQuoteBodySize`.
* **`signedDataToProto`**: Verifies input is at least `quoteV4AuthDataKnownSize` (128 bytes).
* **`certificationDataToProto`**: Verifies input is at least `certificationDataKnownSize` (6 bytes).
* **`qeReportCertificationDataToProto`**: Verifies input is at least `qeReportCertificationDataSignatureEnd` (448 bytes).
* **`enclaveReportToProto`**: Verifies input is at least `qeReportSize` (384 bytes).
* **`qeAuthDataToProto`**: Verifies input is at least `qeAuthDataKnownSize` (2 bytes) and as large as the parsed `authDataEnd` payload.
* **`pckCertificateChainToProto`**: Verifies input is at least `pckCertificateChainKnownSize` (6 bytes).

### 2. Safe Integer Comparisons
Updated size comparisons to use `uint64` or `int64` instead of downcasting to `uint32` or `int32` to prevent potential integer truncation or overflow vulnerabilities when checking untrusted size fields in:
* **`quoteToProtoV4`** (validating `SignedDataSize`)
* **`quoteToProtoV5`** (validating `signedDataSizeV5` and checking for negative sizes)
* **`certificationDataToProto`** (validating cert size)
* **`checkPCKCertificateChain`** (validating cert chain size)

---

### 3. Comprehensive Unit Tests (`abi/abi_test.go`)
Added 7 new targeted edge case tests to `TestQuoteToProto` to ensure all new boundary and size checks are thoroughly validated:
* **`quote too small to determine format`**: Verifies rejection of slices smaller than the 2-byte version header.
* **`v4 quote too small`**: Verifies rejection of V4 quotes smaller than `QuoteMinSize` (1020 bytes).
* **`v4 quote signed data size too large`**: Verifies rejection when `SignedDataSize` is set to `0xffffffff` (exceeding remaining bytes).
* **`v5 quote signed data size too large`**: Verifies rejection when V5 signed data size exceeds remaining bytes.
* **`v5 quote signed data size negative`**: Verifies rejection when V5 signed data size is negative (MSB set, e.g., `0xffffffff`).
* **`v4 quote invalid TEE type in header`**: Verifies rejection of quotes with an invalid `TeeType` in the header.
* **`v4 quote signed data size too small for auth data`**: Verifies rejection when the signed data size is smaller than the minimum required for signature and key (128 bytes).
* **`v4 quote signed data size too small for cert data`**: Verifies rejection when the signed data size leaves insufficient bytes for the certification data header.
* **`v4 quote invalid certification data type`**: Verifies rejection of unsupported certification data types.
* **`v4 quote rawCertificateData too small for QE report`**: Verifies rejection when the remaining bytes are too small to contain a valid QE report.
* **`v4 quote QE auth data too small`**: Verifies rejection when remaining bytes are insufficient to hold the QE auth data header.
* **`v4 quote QE auth data parsed size mismatch`**: Verifies rejection when the QE auth data's `ParsedDataSize` field is modified to exceed the actual remaining bytes.

---

## How was this tested?

**Unit Tests**: Ran `go test ./...` in the workspace. All tests (including the new boundary check tests) passed successfully.
**E2E tests**: Ran attest and check CLI in a prod TDX CVM, TDX verification successfully passed.

### CODE REVIEW

File: abi/abi.go [hunk 1]
```diff
@@ -460,6 +472,12 @@ func tdQuoteBodyDescriptorToProto(b []uint8) (*pb.TDQuoteBodyDescriptor, int32,
 	if tdQuoteBodyDescriptor.TdQuoteBodyType == tdxVersion15BodyType && tdQuoteBodyDescriptor.TdQuoteBodySize != tdQuoteBodySizeV5TDX15 {
 		return nil, 0, fmt.Errorf("TD quote body size is %d, a V5 TDX1.5 quote should have size %d", tdQuoteBodyDescriptor.TdQuoteBodySize, tdQuoteBodySizeV5TDX15)
 	}
+	if tdQuoteBodyDescriptor.TdQuoteBodyType != tdxVersion10BodyType && tdQuoteBodyDescriptor.TdQuoteBodyType != tdxVersion15BodyType {
+		return nil, 0, fmt.Errorf("parsing TD Quote Body Descriptor failed: unsupported TD quote body type , got %d", tdQuoteBodyDescriptor.TdQuoteBodyType)
```
Comment (jkl73) [inline abi/abi.go:475]: extra space

File: abi/abi.go [hunk 2]
```diff
@@ -395,6 +398,9 @@ func quoteToProtoV5(b []uint8) (*pb.QuoteV5, error) {
 
 func headerToProto(b []uint8) (*pb.Header, error) {
 	data := clone(b) // Created an independent copy to make the interface less error-prone
+	if len(data) < headerSize {
```
Comment (jkl73) [inline abi/abi.go]: (here and below) should we make sure len(data) == size?

**Review by jkl73 — APPROVED** [review]

File: abi/abi.go (hunk 1 above)
Comment (yawangwang) [reply abi/abi.go:475]: ah not sure if this is an actual extra space, but I've removed extra spaces somewhere else in this file.

File: abi/abi.go (hunk 2 above)
Comment (yawangwang) [reply abi/abi.go]: yes for some checks we can do that, however for some checks such as `qeReportCertificationDataToProto` we cannot because some parsed data is variable-length. The minimum size check would be sufficient.

**Review by yanchengchen7 — APPROVED** [review]

File: abi/abi.go [hunk 3]
```diff
@@ -460,6 +471,12 @@ func tdQuoteBodyDescriptorToProto(b []uint8) (*pb.TDQuoteBodyDescriptor, int32,
 	if tdQuoteBodyDescriptor.TdQuoteBodyType == tdxVersion15BodyType && tdQuoteBodyDescriptor.TdQuoteBodySize != tdQuoteBodySizeV5TDX15 {
 		return nil, 0, fmt.Errorf("TD quote body size is %d, a V5 TDX1.5 quote should have size %d", tdQuoteBodyDescriptor.TdQuoteBodySize, tdQuoteBodySizeV5TDX15)
 	}
+	if tdQuoteBodyDescriptor.TdQuoteBodyType != tdxVersion10BodyType && tdQuoteBodyDescriptor.TdQuoteBodyType != tdxVersion15BodyType {
+		return nil, 0, fmt.Errorf("parsing TD Quote Body Descriptor failed: unsupported TD quote body type , got %d", tdQuoteBodyDescriptor.TdQuoteBodyType)
+	}
+	if len(data) < int(tdQuoteBodyDescriptor.TdQuoteBodySize) {
```
Comment (yanchengchen7) [inline abi/abi.go]: nit: len(data) < quoteBodyTypeSizeV5 + quoteBodySizeFieldLengthV5 + int(tdQuoteBodyDescriptor.TdQuoteBodySize)?

Comment (yawangwang) [reply abi/abi.go]: good catch, done. Also removed some redundant checks.
